File: src/main/java/br/com/github/williiansilva51/finfrases/domain/Quote.java
package br.com.github.williiansilva51.finfrases.domain;

import br.com.github.williiansilva51.finfrases.domain.enums.CategoryQuote;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "quotes")
public class Quote {
    @Id
    private String id;

    @Field("content")
    private String content;

    @Field("author")
    @Indexed
    private String author;

    @Field("category")
    @Indexed
    private List<CategoryQuote> tags;

    @Field("source")
    private String source;

    @Field("verified")
    private boolean verified;

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: api/models/quote.py
from datetime import datetime, timezone
from typing import Annotated

from beanie import Document, Indexed
from pydantic import Field

from .enums import CategoryQuote


class Quote(Document):
    content: str
    author: Annotated[str, Indexed()]
    tags: list[CategoryQuote]
    source: str | None
    verified: bool = False
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "quotes"
